pkg/parse/parse.go

```diff
@@ -333,27 +333,28 @@ func getMounts(mounts []string) (map[string]specs.Mount, error) {
 }
 
 // GetBindMount parses a single bind mount entry from the --mount flag.
-func GetBindMount(args []string) (specs.Mount, error) {
+func GetBindMount(args []string, contextDir string) (specs.Mount, error) {
 	newMount := specs.Mount{
 		Type: TypeBind,
 	}
 
-	setSource := false
 	setDest := false
+	bindNonRecursive := false
 
 	for _, val := range args {
 		kv := strings.SplitN(val, "=", 2)
 		switch kv[0] {
 		case "bind-nonrecursive":
 			newMount.Options = append(newMount.Options, "bind")
-		case "ro", "nosuid", "nodev", "noexec":
+			bindNonRecursive = true
+		case "ro", "nosuid", "nodev", "noexec", "rw", "readwrite":
```

Comment:
Our `chroot` isolation implementation doesn't recognize either "rw" or "readwrite", and a peek at runc and crun sources suggests that they don't recognize "readwrite".

Reply:
@nalind I think crun supports `rw` https://github.com/containers/crun/blob/main/src/libcrun/linux.c#L378

Reply:
But we need those for the feature to be complete, don't we?

Reply:
Could we just handle it here since `rw` == `readwrite`  instead of offloading to runtime.

Reply:
I'll check the relevant changes which needs to be done for `chroot` isolation part.

Reply:
@nalind Just curious is `rw` a valid use-case for isolated roots since behavior is expected to be immutable.

Reply:
The [doc](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypebind-the-default-mount-type) says the bind mount is read-only by default, but writeable is an option.

Reply:
`rw` is working as expected for both regular builds and isolated roots. I have also written a test for the same.